### ISSUE TITLE
osisRef string for a verse range in crossReference notes

### ISSUE THREAD

**Issue by DavidHaslam** [krisek/HunKar#12]
There are 3468 matches to the regexp `,\d+-\d+` each of which is in the _vernacular_ part of a **reference** element.

In each case the **osisRef** string points only to the **leading** verse of the **verse range**.

This is not the correct method to mark a **verse range** in such a reference.

Please refer to the wiki page [OSIS Bibles](https://wiki.crosswire.org/OSIS_Bibles#Marking_cross-references_notes).

The last reference line has this example for a **verse range**.
```
<reference osisRef="Ezek.36.26-Ezek.36.27">36:26, 27</reference>
```
Although this English Bible illustration had a _comma_ to separate two consecutive verses in the _vernacular_, it's still treated correctly as a verse range.

```
osisRef="Ezek.36.26-Ezek.36.27"
```
The same method should be used for all the **verse range** references in the XML file.

NB. There are some other patterns that may require further investigation.
```
<reference osisRef="Joel.2.6">Jóel 2 6. Jóel,2,6. 7</reference>
```
```
<reference osisRef="Amos.8.6">Ámós 8 6. Ámós,2,6. 7</reference>
```
_You may need to dig deeper to discover all the problematic references._

**Comment by DavidHaslam** [comment]
@krisek 

Special care is required for parsing more complex references. 

Here's an example found in `Joel.2.30`:
```
<reference osisRef="Joel.9.2">Jóel 9,2-4.10</reference>
```
This is a _vernacular_ reference to verses 2-4 & 10 in Joel 9.

It should be split into two _separate_ **reference** elements - with the separator period _between_ them.
```
<reference osisRef="Joel.9.2-Joel.9.4">Jóel 9,2-4</reference>.<reference osisRef="Joel.9.10">10</reference>
```
It is not such a simple task to parse and process this correctly. There must be many more like it.

**Comment by krisek** [comment]
I tried to fix all in 6ac5d63, I wonder what should be the 'close' criteria for this... I think I did everything what is realistically possible in a release cycle

**Comment by krisek** [comment]
I close the issue, the verse ranges should be ok now. We open new ones if issues found with particular references.
